### PR TITLE
Remove `strong_typedef` in `fixed_point.hpp`

### PR DESCRIPTION
**Pull request by codereport**
With C++17's direct-list-initialization for enumerators, we can remove the `strong_typedef` that is used for `scale_type` in the `fixed_point` code. This PR makes that change. For more info see: https://en.cppreference.com/w/cpp/language/enum

### CODE REVIEW

File: cpp/include/cudf/fixed_point/fixed_point.hpp [hunk 1]
```diff
@@ -31,20 +31,8 @@
...
-template <typename T>
-struct strong_typedef {
-  T _t;
-  CUDA_HOST_DEVICE_CALLABLE explicit constexpr strong_typedef(T t) : _t(t) {}
-  CUDA_HOST_DEVICE_CALLABLE operator T() const { return _t; }
-};
-/** \endcond */
 
-using scale_type = strong_typedef<int32_t>;
+enum scale_type : int32_t {};
```
Comment (ttnghia) [inline cpp/include/cudf/fixed_point/fixed_point.hpp:35]:
```suggestion
enum class scale_type : int32_t {};
```

Does this enum contain any enum value?

**Review by ttnghia — CHANGES_REQUESTED** [review]

File: cpp/include/cudf/fixed_point/fixed_point.hpp (hunk 1 above)
Comment (codereport) [reply cpp/include/cudf/fixed_point/fixed_point.hpp:35]: For our purposes, we want implicit conversions. What we want is enforcing the "strong typing" at the call site. See here: https://www.godbolt.org/z/b38xdTqoc. As for the second question, this is just the pattern of how to use direct-list-initialiation for "strong typing".

Comment (vuule) [reply cpp/include/cudf/fixed_point/fixed_point.hpp:35]: > For our purposes, we want implicit conversions. 

So a scale_type variable has to be explicitly typed to create, but it implicitly casts to an int. Is this correct?

**Review by vuule — APPROVED** [review]

File: cpp/include/cudf/fixed_point/fixed_point.hpp (hunk 1 above)
Comment (codereport) [reply cpp/include/cudf/fixed_point/fixed_point.hpp:35]: Yep!

**Review by ttnghia — APPROVED** [review]

**Comment by harrism** [comment]
@codereport looks like there are more places to update: 

```
06:57:53 /workspace/.conda-bld/cudf_1619469958526/work/cpp/include/cudf/fixed_point/fixed_point.hpp: In function ‘auto numeric::detail::negate(const numeric::scale_type&)’:
06:57:53 /workspace/.conda-bld/cudf_1619469958526/work/cpp/include/cudf/fixed_point/fixed_point.hpp:102:58: error: invalid conversion from ‘int’ to ‘numeric::scale_type’ [-fpermissive]
06:57:53  auto negate(scale_type const& scale) { return scale_type{-scale}; }

**Comment by codereport** [comment]
> @codereport looks like there are more places to update:
> 
> ```
> 06:57:53 /workspace/.conda-bld/cudf_1619469958526/work/cpp/include/cudf/fixed_point/fixed_point.hpp: In function ‘auto numeric::detail::negate(const numeric::scale_type&)’:
> 06:57:53 /workspace/.conda-bld/cudf_1619469958526/work/cpp/include/cudf/fixed_point/fixed_point.hpp:102:58: error: invalid conversion from ‘int’ to ‘numeric::scale_type’ [-fpermissive]
> 06:57:53  auto negate(scale_type const& scale) { return scale_type{-scale}; }
> ```

I am currently looking into this. It compiles locally and should not be an error FWICT: https://www.godbolt.org/z/a99s45zPj

**Comment by codereport** [comment]
@gpucibot merge

**Pull request merged**